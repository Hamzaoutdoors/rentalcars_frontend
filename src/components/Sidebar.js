/* eslint-disable max-len */
import React from 'react';
import { NavLink } from 'react-router-dom';
import {
  CDBSidebar, CDBSidebarHeader, CDBSidebarMenuItem, CDBSidebarContent, CDBSidebarMenu, CDBSidebarFooter,
} from 'cdbreact';
import {
  Facebook, Instagram, Pinterest, Twitter,
} from '@material-ui/icons';
import styled from 'styled-components';
import { desktop, mobile } from '../responsive';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
  overflow: scroll-initial;
  border-right: 1px solid #e6e6e6;
  color: black;
  align-items: center;
`;

const Logo = styled(NavLink)`
  height: 200px;
  width: 200px;
  margin-bottom: 10px;
  text-decoration: none !important;
  ${mobile({
    display: 'none',
  })};
`;

const LogoImage = styled.img`
  width: 100%;
  height: 100%;
`;

const Wrapper = styled.div`
  color: black;
  background-color: white;
    ${mobile({
    width: '10px !important',
  })};
`;

const Header = styled(CDBSidebarHeader)`
    ${desktop({
    display: 'none',
  })};
`;

const Content = styled.div`
    padding: 0px;
    margin: 0px;
    display: flex;
    flex-direction: column;
    height: 100%;
    width: 100%;
    justify-content: space-between;
`;

const LinkElem = styled(NavLink)`
  font-size: 16px;
  font-weight: 400;
  font-weight: bold;
`;

const SidebarItem = styled(CDBSidebarMenuItem)`
  color: black;
  text-decoration: none;
  margin-left: 30px;
  text-transform: uppercase;
  &.activeClicked {
    background-color: #f5f5f5;
}
`;

const SocialContainer = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    padding: '20px 5px';
    text-align: center;
`;

const SocialIcon = styled.div`
   width: 25px;
   height: 25px;
   border-radius: 50%;
   color: #fff;
   background-color : #${(props) => props.color};
   cursor: pointer;
   display: flex;
   justify-content: center;
   align-items: center;
   margin-right: 20px;
`;

const Signature = styled.span`
    font-size: 12px;
    font-weight: 400;
    text-align: center;
`;

const Sidebar = () => (
  <Container>
    <Logo to="/">
      <LogoImage src="https://i.ibb.co/vxkd1PT/LOGO.png" />
    </Logo>
    <Wrapper as={CDBSidebar}>
      <Header prefix={<i className="fa fa-bars fa-large fs-3" />} />
      <Content as={CDBSidebarContent} className="sidebar-content">
        <CDBSidebarMenu>
          <LinkElem
            exact
            to="/cars"
          >
            <SidebarItem icon="columns" ac>
              Models
            </SidebarItem>
          </LinkElem>
          <LinkElem exact to="/lifestyle" activeClassName="activeClicked">
            <SidebarItem icon="table">
              LIFESTYLE
            </SidebarItem>
          </LinkElem>
          <LinkElem exact to="/" activeClassName="activeClicked">
            <SidebarItem icon="shopping-cart">
              shop
            </SidebarItem>
          </LinkElem>
          <LinkElem exact to="/dashboard" activeClassName="activeClicked">
            <SidebarItem icon="car">
              test drive
            </SidebarItem>
          </LinkElem>
        </CDBSidebarMenu>
      </Content>
      <CDBSidebarFooter style={{ textAlign: 'center' }}>
        <SocialContainer className="sidebar-btn-wrapper">
          <SocialIcon color="3b5999">
            <Facebook />
          </SocialIcon>
          <SocialIcon color="e4405f">
            <Instagram />
          </SocialIcon>
          <SocialIcon color="55acee">
            <Twitter />
          </SocialIcon>
          <SocialIcon color="e60023">
            <Pinterest />
          </SocialIcon>
        </SocialContainer>
        <Signature>
          2022 Â© All rights reserved
        </Signature>
      </CDBSidebarFooter>
    </Wrapper>
  </Container>
);

export default Sidebar;
