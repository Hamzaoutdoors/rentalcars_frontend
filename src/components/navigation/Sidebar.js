/* eslint-disable no-unused-vars */
import { React, useEffect } from 'react';
import { useSelector } from 'react-redux';
import DesktopBar from './desktop/DesktopBar';
import MobileBar from './mobile/MobileBar';
import store from '../../redux/configureStore';
import { toggleMobile, toggleMobileModal } from '../../redux/utils/actions/navActions';

const setDefaultWhenUnmount = () => {
  store.dispatch(toggleMobile(false));
  store.dispatch(toggleMobileModal(false));
};

const handleResize = () => {
  if (window.innerWidth > 1150) {
    store.dispatch(toggleMobile(false));
  } else {
    store.dispatch(toggleMobile(true));
  }
};

const Sidebar = () => {
  const { mobile } = useSelector((state) => state.utils.navBar);
  const { isAuthenticated } = useSelector((state) => state.auth);

  useEffect(() => {
    handleResize();

    window.addEventListener('resize', () => {
      handleResize();
    });

    return () => {
      setDefaultWhenUnmount();
    };
  }, []);

  if (mobile) {
    return (
      <>
        <MobileBar />
      </>
    );
  }

  return (
    <>
      <DesktopBar />
    </>
  );
};

export default Sidebar;
