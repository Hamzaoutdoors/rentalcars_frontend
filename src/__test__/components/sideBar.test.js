import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Sidebar from '../../components/navigation/Sidebar';

it('Sidebar component renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Sidebar />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
