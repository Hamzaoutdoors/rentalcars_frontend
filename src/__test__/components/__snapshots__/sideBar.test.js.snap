// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders correctly 1`] = `
<div
  className="sc-iqseJM jkDKGn"
>
  <a
    aria-current="page"
    className="sc-crHmcD jnOiEo active"
    href="/"
    onClick={[Function]}
  >
    <img
      className="sc-egiyK ZGZjt"
      src="https://i.ibb.co/vxkd1PT/LOGO.png"
    />
  </a>
  <div
    className="sc-pVTFL jXOalY"
  >
    <div
      className="sc-jrQzAO iuYxOP sidebar-content"
    >
      <a
        className="sc-iCfMLu cXqcxp"
        href="/cars"
        onClick={[Function]}
      >
        Reserve a Car
      </a>
      <a
        className="sc-iCfMLu cXqcxp"
        href="/lifestyle"
        onClick={[Function]}
      >
        LifeStyle
      </a>
      <a
        className="sc-iCfMLu cXqcxp"
        href="/login"
        onClick={[Function]}
      >
        Log In
      </a>
      <a
        className="sc-iCfMLu cXqcxp"
        href="/sign_up"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </div>
  <footer
    className="sc-hKwDye dNJIvQ"
  >
    <div
      className="sc-bdvvtL cMRNAV sidebar-btn-wrapper"
    >
      <div
        className="sc-gsDKAQ hzahuo"
        color="3b5999"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
          />
        </svg>
      </div>
      <div
        className="sc-gsDKAQ OJySL"
        color="e4405f"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
          />
        </svg>
      </div>
      <div
        className="sc-gsDKAQ kWbjn"
        color="55acee"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
          />
        </svg>
      </div>
      <div
        className="sc-gsDKAQ FQiJf"
        color="e60023"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9.04 21.54c.96.29 1.93.46 2.96.46a10 10 0 0 0 10-10A10 10 0 0 0 12 2 10 10 0 0 0 2 12c0 4.25 2.67 7.9 6.44 9.34-.09-.78-.18-2.07 0-2.96l1.15-4.94s-.29-.58-.29-1.5c0-1.38.86-2.41 1.84-2.41.86 0 1.26.63 1.26 1.44 0 .86-.57 2.09-.86 3.27-.17.98.52 1.84 1.52 1.84 1.78 0 3.16-1.9 3.16-4.58 0-2.4-1.72-4.04-4.19-4.04-2.82 0-4.48 2.1-4.48 4.31 0 .86.28 1.73.74 2.3.09.06.09.14.06.29l-.29 1.09c0 .17-.11.23-.28.11-1.28-.56-2.02-2.38-2.02-3.85 0-3.16 2.24-6.03 6.56-6.03 3.44 0 6.12 2.47 6.12 5.75 0 3.44-2.13 6.2-5.18 6.2-.97 0-1.92-.52-2.26-1.13l-.67 2.37c-.23.86-.86 2.01-1.29 2.7v-.03z"
          />
        </svg>
      </div>
    </div>
    <span
      className="sc-dkPtRN lmkHJB"
    >
      2022 Â© MIT license
    </span>
  </footer>
</div>
`;
